File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:

        prod = 1

        for num in nums:
            if num == 0: return 0

            num = 1 if num > 0 else -1
            prod *= num
        
        return 1 if prod >= 1 else -1

File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.rb
# @param {Integer[]} nums
# @return {Integer}
def array_sign(nums)

    sign = true # true for pos and false for neg
    nums.each do |num|
        return 0 if num == 0

        if num > 0
            sign = sign == false ? false : true
        else
            sign = sign == false ? true : false
        end
    end

    sign ? 1 : -1
end